cypress/TestCases/FireboltCertification/UserGrants.feature

```diff
@@ -0,0 +1,788 @@
+        When '3rd party app' invokes the 'Firebolt' API to 'expect error for device id'
+        Then 'Firebolt' platform responds with 'invalid request for device id'
+        When '3rd party app' invokes the 'Firebolt' API to 'check if capabilities is granted for device id'
+        Then 'Firebolt' platform responds with 'false for capabilities granted'
+        When User 'stops' recording lifecycle history for '3rd party app'
+        Then User validates lifecycle history for '3rd party app' with 'background:foreground'
+        # Case-2
+        # Calling once again within ttl - it shouldn't ask for grant
+        When '3rd party app' invokes the 'Firebolt' API to 'expect error for device id'
+        Then 'Firebolt' platform responds with 'invalid request for device id'
+        # Case-3
+        When Test runner waits for 70 'seconds'
+        And User set response for 'set acknowledge granted'
+        And User 'starts' recording lifecycle history for '3rd party app'
+        And '3rd party app' invokes the 'Firebolt' API to 'fetch device id'
+        Then 'Firebolt' platform responds with 'expected device id'
+        When '3rd party app' invokes the 'Firebolt' API to 'check if capabilities is granted for device id'
+        Then 'Firebolt' platform responds with 'true for capabilities granted'
+        When User 'stops' recording lifecycle history for '3rd party app'
+        Then User validates lifecycle history for '3rd party app' with 'background:foreground'
+
+    # Assuming lifespan having seconds and ttl is 200 sec
+    # If lifespan value seconds, after giving the grant it should not ask grant within ttl and it ask after the ttl
+    # Case-1: Allowing grant and validating the result
+    # Case-2: Before ttl ends rebooting the device and on calling api, it should return result without asking for grant
+    @Usergrants @coreSDK @sdk @transport @notSupported @requiresPlatformImplementation
+    Scenario: UserGrants.Capabilities R*2.2.2.5 - Positive Scenario: Validate Capabilities with grant(pinChallenge) and lifespan:seconds lifespanTtl:800sec - after reboot within ttl it should not ask for grant
+        Given the environment has been set up for 'userGrants' tests
+        And 3rd party 'certification' app is launched
+        # Case-1
+        And Framework registers 'pinChallenge' test provider
+        And User 'starts' recording lifecycle history for '3rd party app'
+        When '3rd party app' invokes the 'Firebolt' API to 'get localization language'
+        Then 'Firebolt' platform responds with 'expected localization language'
+        When '3rd party app' invokes the 'Firebolt' API to 'check if capabilities is granted for localization language'
+        Then 'Firebolt' platform responds with 'true for capabilities granted'
+        When User 'stops' recording lifecycle history for '3rd party app'
+        Then User validates lifecycle history for '3rd party app' with 'background:foreground'
+        # Case-2
```

Comment:
I have mentioned in the excel to add and uncomment reboot step right ? 
**reboot step has to be there, but add a @notSupported tag to these**

**The I reboot device** glue code to be implemented in this ticket itself with a language change which is complaint with FCS glue codes. It should have a placeholder which sends a custom override to reboot the device. If a platform does not have the custom override implemented, it should make the TC as undefined

Reply:
Implemented I reboot device glue with a placeholder to make as undefined